app/scripts/router.js

```diff
@@ -244,9 +247,18 @@ function (
       // rendered for any reason, including if the view was
       // automatically redirected.
       var self = this;
-
       viewToShow.logScreen();
-      return viewToShow.render()
+      // perform any actions needed by the broker before the view has rendered
+      return self.broker.beforeRender()
+        .then(
+          function () {
+            return viewToShow.render();
+          }, function () {
+            // log the timeout event, this even should only happen in Firefox <43
+            viewToShow.logScreenEvent('broker.timeout');
```

Comment:
This was my thought too, I would pass an [AuthErrors.DESKTOP_CHANNEL_TIMEOUT](https://github.com/mozilla/fxa-content-server/blob/5cf3012dc63bf38e6be97d9c2802debffa27a632/app/scripts/lib/auth-errors.js#L173) or a more generic [AuthErrors.CHANNEL_TIMEOUT](https://github.com/mozilla/fxa-content-server/blob/5cf3012dc63bf38e6be97d9c2802debffa27a632/app/scripts/lib/auth-errors.js#L189)